src/sentry/integrations/utils/codecov.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional, Sequence, Tuple
+
+import requests
+
+from sentry import options
+
+CODECOV_URL = "https://api.codecov.io/api/v2/{service}/{owner_username}/repos/{repo_name}/report"
+CODECOV_TOKEN = options.get("codecov.client-secret")
+
+
+class Codecov:
+    def get_codecov_line_coverage(
+        repo: str, service: str, branch: str, path: str
+    ) -> Tuple[Optional[Sequence[Tuple[int, int]]], Optional[int]]:
+        owner_username, repo_name = repo.split("/")
+        if service == "github":
+            service = "gh"
+        url = CODECOV_URL.format(
+            service=service, owner_username=owner_username, repo_name=repo_name
+        )
+        line_coverage, status_code = None, None
+        params = {"branch": branch, "path": path}
+        response = requests.get(
+            url, params=params, headers={"Authorization": f"tokenAuth {CODECOV_TOKEN}"}
+        )
+        line_coverage, status_code = (
+            response.json()["files"][0]["line_coverage"],
+            response.status_code,
+        )
+
+        return line_coverage, status_code
```

Comment:
Use `response.raise_for_status()` instead of returning two values.

Reply:
I see why this would be beneficial for code cleanliness, but when @snigdhas and I discussed how we wanted to receive the status code on the FE, we chose a separate field in `result[codecovStatusCode]`. This already is being used in the [FE](https://github.com/getsentry/sentry/blob/851fc2b61e240ce6ba56088cd0100f72737cb698/static/app/types/integrations.tsx#L153-L157), and will cause extra work to change it.

Would it be okay if we kept it like this for now, and fixed this after if we have time?